volume-providers/vpc/provider/util.go

```diff
@@ -86,6 +86,40 @@ func skipRetry(err *models.Error) bool {
 	return false
 }
 
+// FlexyRetry ...
+func FlexyRetry(logger *zap.Logger, retryfunc func() (interface{}, error), skipRetryFunc func(interface{}, *models.Error) bool) error {
```

Comment:
I would encapsulate` FlexyRetry` method in `FlexyRetry` struct as below
```type FlexyRetry {
int maxRetryAttempt
int maxrRetryGap
}```

With this , caller can create  NewFlexyRetry(maxRetryAttempt,maxrRetryGap) if global(vpc config)  retry parameters are not suitable for his requirement. Ther will be 2 method 
NewFlexyRetryDefault()  -- Create FlexyRetry from confg . This will be applicable for most of the cases 
NewFlexyRetry(maxRetryAttempt,maxrRetryGap)  - Create  FlexyRetry from given parameters

Reply:
done!